src/core/trtserver.h

```diff
@@ -376,6 +377,96 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceNew(
 /// \return a TRTSERVER_Error indicating success or failure.
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceDelete(TRTSERVER_Trace* trace);
 
+/// Get the activity userp of the trace object, which is supplied in the call to
+/// TRTSERVER_TraceNew.
+/// \param trace The trace object.
+/// \param userp Returns the activity userp of the trace object.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceUserp(
+    TRTSERVER_Trace* trace, void** userp);
+
+/// Get the name of the model being traced. The caller
+/// does not own the returned string and must not modify or delete
+/// it. The lifetime of the returned string extends only as long as
+/// 'trace' and must not be accessed once 'trace' is deleted.
+/// 'model_name' is guaranteed to be valid at invocation of
```

Comment:
"... guaranteed to be valid until invocation... " ?

Reply:
This is tricky because these fields, except `id`, are not set when user calls `TRTSERVER_TraceNew`, "until" sounds like the get functions can be called immediately after `TRTSERVER_TraceNew`

Reply:
I see. Then perhaps "This method is only guaranteed to correctly return the 'model_name' in the <trace release callback>."